### PR TITLE
test(entities): fix doctests

### LINKED ISSUE #1254: Fix entities doctests
Part of #1061

### PR DESCRIPTION
**Pull request by bonjourmauko**
Fixes #1254 
Depends on #1253 
Supersedes #1034  

#### New features

- Make `CoreEntity` public
  - Allows for more easily creating customised entities.

#### Technical changes

- Add missing doctests.

### CODE REVIEW

**Comment by bonjourmauko** [comment]
Poke @benjello @benoit-cty

File: openfisca_core/entities/_core_entity.py [hunk 1]
```diff
@@ -72,11 +73,46 @@ def get_variable(
...
+            >>> this.get_variable("tax")
+
+            >>> this.get_variable("tax", check_existence=True)
+            Traceback (most recent call last):
+            VariableNotFoundError: You tried to calculate or to set a value...
+
+            >>> class tax(variables.Variable):
+            ...     definition_period = periods.MONTH
+            ...     value_type = float
+            ...     entity = that
```
Comment (benjello) [inline openfisca_core/entities/_core_entity.py:107]: Why using different entities, `this` and `that`  here ?

File: openfisca_core/entities/_core_entity.py [hunk 2]
```diff
@@ -90,16 +126,47 @@ def check_variable_defined_for_entity(self, variable_name: t.VariableName) -> No
...
+            >>> this.set_tax_benefit_system(tax_benefit_system)
+
+            >>> this.check_variable_defined_for_entity("tax")
+            Traceback (most recent call last):
+            VariableNotFoundError: You tried to calculate or to set a value...
+
+            >>> class tax(variables.Variable):
+            ...     definition_period = periods.WEEK
+            ...     value_type = int
+            ...     entity = that
```
Comment (benjello) [inline openfisca_core/entities/_core_entity.py:153]: Same as before.
Why using different entities, `this` and `that`  here ?

File: openfisca_core/entities/_core_entity.py (hunk 1 above)
Comment (bonjourmauko) [reply openfisca_core/entities/_core_entity.py:107]: To show that this method is going to get you the variable regardless of who is this variable being defined for. It was a confusing behaviour for me (a couple of years ago I implemented an actual filter, to search within the variables defined for the entity, but I'll repropose that one later as it is a breaking change).

File: openfisca_core/entities/_core_entity.py (hunk 2 above)
Comment (bonjourmauko) [reply openfisca_core/entities/_core_entity.py:153]: Same answer as above.

File: openfisca_core/entities/_core_entity.py (hunk 1 above)
Comment (bonjourmauko) [reply openfisca_core/entities/_core_entity.py:107]: (The test in itself has exactly that purpose, for people to think "Wait, this is weird behaviour". If tomorrow someone changes this, the test will fail, and that is what I actually expect).

**Review by benjello — APPROVED** [review]

**Pull request merged**